### PR TITLE
feat(buckets): add pagination limits to prevent performance issues

### PR DESCRIPTION
**Pull request by uipreliga**
- Add PaginationLimitError exception class with unified create() factory method
  - Add _MAX_AUTO_PAGINATION_PAGES=10 class constant
  - Add pagination limits to all auto-paginating methods:
    - list() and list_async() - offset-based (100/50 items per page)
    - list_files() and list_files_async() - token-based (500 items per page)
    - get_files() and get_files_async() - glob pattern-based (500 items per page)
  - Provide context-aware error messages with conditional filter guidance
  - Add comprehensive test coverage (14 tests) for all edge cases

### CODE REVIEW

File: src/uipath/_services/buckets_service.py [hunk 1]
```diff
@@ -63,8 +70,9 @@ def list(
         """
         skip = 0
         top = 100
+        pages_fetched = 0
 
-        while True:
+        while pages_fetched < self._MAX_AUTO_PAGINATION_PAGES:
```
Comment (AlexAndriesUiPath) [inline src/uipath/_services/buckets_service.py]: why don't we allow the consumer to decide what page can fetch by allowing to pass a `skip` and `top` arguments?
we shouldn't limit the number of results consumer can get

Comment (cristipufu) [reply src/uipath/_services/buckets_service.py]: I 💯 vote against having pagination inside these service methods. Let the caller do the pagination, restrict to max x items per page

File: src/uipath/_services/buckets_service.py [hunk 2]
```diff
@@ -99,40 +134,79 @@ async def list_async(
...
+            ...         break
+            ...     skip += top
+        """
+        # Validate parameters
+        if skip < 0:
+            raise ValueError("skip must be >= 0")
+        if top < 1:
+            raise ValueError("top must be >= 1")
+        if top > 1000:
+            warnings.warn(
```
[the remark this answers is not in the document]
Comment (uipreliga) [reply src/uipath/_services/buckets_service.py]: Fixed

File: src/uipath/_services/buckets_service.py [hunk 3]
```diff
@@ -37,60 +40,100 @@ def list(
...
-                name=name,
-                skip=skip,
-                top=top,
+        # Validate parameters
+        if skip < 0:
+            raise ValueError("skip must be >= 0")
+        if skip > MAX_SKIP_OFFSET:
+            raise ValueError(
+                f"skip must be <= {MAX_SKIP_OFFSET} (requested: {skip}). "
+                f"For large datasets, use list_files() with continuation tokens instead of offset-based pagination."
```
Comment (AlexAndriesUiPath) [inline src/uipath/_services/buckets_service.py]: any reason why we accept a specific max skip value? performance issue or why?

**Review by emanueldejanu — APPROVED** [review]
